### PR TITLE
Use setter to replace kvo for filelogger

### PR DESCRIPTION
**Pull request by kinarobin**
### New Pull Request Checklist

* [x] I have read and understood the [CONTRIBUTING guide](https://github.com/CocoaLumberjack/CocoaLumberjack/blob/master/.github/CONTRIBUTING.md)
* [x] I have read the [Documentation](http://cocoadocs.org/docsets/CocoaLumberjack/)
* [x] I have searched for a similar pull request in the [project](https://github.com/CocoaLumberjack/CocoaLumberjack/pulls) and found none

* [x] I have updated this branch with the latest master to avoid conflicts (via merge from master or rebase)
* [x] I have added the required tests to prove the fix/feature I am adding
* [x] I have updated the documentation (if necessary)
* [x] I have run the tests and they pass
* [x] I have run the lint and it passes (`pod lib lint`)

This merge request fixes / refers to the following issues: ... #463 

### Pull Request Description

`maximumNumberOfLogFiles` & `logFilesDiskQuota` of `DDLogFileManagerDefault` is not KVO, they are disabled intentionally.

 So it's a bug for user to reset `maximumNumberOfLogFiles` & `logFilesDiskQuota`. 

With KVO crash, I think `setter` is more better!

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -14,6 +14,7 @@
 - Add backend for swift-log (#1164)
 - Specify CocoaPods version to ensure `swift_version` attribute works (#1167)
 - Simplify `DDLogFileManager` callbacks for archived log files (#1166)
+- Use setter to replace kvo for filelogger (#1180)
```
Comment (ffried) [inline CHANGELOG.md:17]: This is in the wrong section. 3.7.0 has already been released. Please move it up into the 3.7.1 section.

File: Sources/CocoaLumberjack/DDFileLogger.m [hunk 2]
```diff
@@ -129,43 +116,32 @@ - (instancetype)initWithLogsDirectory:(NSString *)logsDirectory
...
 #endif
 
-- (void)dealloc {
-    // try-catch because the observer might be removed or never added. In this case, removeObserver throws an exception
-    @try {
-        [self removeObserver:self forKeyPath:NSStringFromSelector(@selector(maximumNumberOfLogFiles))];
-        [self removeObserver:self forKeyPath:NSStringFromSelector(@selector(logFilesDiskQuota))];
-    } @catch (NSException *exception) {
-    }
+- (void)_deleteOldLogFiles {
```
Comment (ffried) [inline Sources/CocoaLumberjack/DDFileLogger.m:119]: I think the name should contain the reason why we don't use `-deleteOldFiles` directly. Something like `deleteOldFilesForConfigurationChange` or even `configurationChanged` would make it more clear what the method does.

File: Sources/CocoaLumberjack/DDFileLogger.m [hunk 3]
```diff
@@ -129,43 +116,32 @@ - (instancetype)initWithLogsDirectory:(NSString *)logsDirectory
...
-        });
+- (void)setMaximumNumberOfLogFiles:(NSUInteger)maximumNumberOfLogFiles {
+    if (_maximumNumberOfLogFiles != maximumNumberOfLogFiles) {
+        _maximumNumberOfLogFiles = maximumNumberOfLogFiles;
+        NSLogInfo(@"DDFileLogManagerDefault: Responding to configuration change: maximumNumberOfLogFiles");
+        [self _deleteOldLogFiles];
     }
 }
 
+
```
Comment (ffried) [inline Sources/CocoaLumberjack/DDFileLogger.m:144]: NitPick: unnecessary newline

**Review by ffried — CHANGES_REQUESTED** [review]
Thanks for this change! Generally looks good. Just a few small things.

File: CHANGELOG.md (hunk 1 above)
Comment (kinarobin) [reply CHANGELOG.md]: Oh, my mistake.

File: Sources/CocoaLumberjack/DDFileLogger.m (hunk 2 above)
Comment (kinarobin) [reply Sources/CocoaLumberjack/DDFileLogger.m]: You are right.

File: Sources/CocoaLumberjack/DDFileLogger.m (hunk 3 above)
Comment (kinarobin) [reply Sources/CocoaLumberjack/DDFileLogger.m]: Done!

**Review by ffried — APPROVED** [review]
LGTM now. Thanks!

**Pull request merged**